float32 distance
---
bool success
string message